lms/djangoapps/mailing/management/commands/mailchimp_sync_announcements.py

```diff
@@ -0,0 +1,69 @@
+import logging
+import math
+import random
+import itertools
+from itertools import chain
+from optparse import make_option
+from collections import namedtuple
+
+
+from django.core.management.base import BaseCommand, CommandError
+
+from mailsnake import MailSnake
+
+from django.contrib.auth.models import User
+
+from .mailchimp_sync_course import (connect_mailchimp, get_subscribed,
+                                    get_unsubscribed, get_cleaned,
+                                    subscribe_with_data)
+
+log = logging.getLogger('edx.mailchimp')
+
+
+class Command(BaseCommand):
+    args = '<mailchimp_key mailchimp_list course_id>'
+    help = 'Synchronizes a mailchimp list with the students of a course.'
+
+    option_list = BaseCommand.option_list + (
+        make_option('--key', action='store', help='mailchimp api key'),
+        make_option('--list', action='store', dest='list_id',
+                    help='mailchimp list id'),
+    )
+
+    def parse_options(self, options):
+        if not options['key']:
+            raise CommandError('missing key')
+
+        if not options['list_id']:
+            raise CommandError('missing list id')
+
+        return (options['key'], options['list_id'])
+
+    def handle(self, *args, **options):
+        key, list_id = self.parse_options(options)
+
+        log.info('Syncronizing announcement mailing list')
+
+        mailchimp = connect_mailchimp(key)
+
+        subscribed = get_subscribed(mailchimp, list_id)
+        unsubscribed = get_unsubscribed(mailchimp, list_id)
+        cleaned = get_cleaned(mailchimp, list_id)
+        non_subscribed = unsubscribed.union(cleaned)
+
+        enrolled = get_enrolled()
```

Comment:
Do we use this `enrolled` anywhere else (doesn't look like it) - if so I wonder why the `get_enrolled` query grabs the user objects -- seems like it would be simpler to get just the email at the same time the query is done, so we don't spend the extra time at L68 grabbing the email address off ALL the user objects.